cmd/nerdctl/build.go

```diff
@@ -146,6 +149,17 @@ func buildAction(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	iidFile, _ := cmd.Flags().GetString("iidfile")
+	if iidFile != "" {
+		id, err := getDigestFromMetaFile(iidFile)
```

Comment:
ID can be fetched from loadImage() with a modification like this https://github.com/containerd/nerdctl/pull/582/files#diff-714eb9b930b7e765168c40536f11caba9f7c5e8c9d0f097c2c0da77842a53ae6R183

Reply:
`--output=type=image --iidfile=id`

`loadImage` should not be called when `output=type=image`, will raise `unrecognized image format` error.